### PR TITLE
Adding a hook from Ryan Florence's 2018 React Conf talk.

### PR DESCRIPTION
**Pull request by JHughes11**
edit: Adds to #1

### CODE REVIEW

File: hooks.js [hunk 1]
```diff
@@ -151,5 +151,51 @@ export const hooks = [
     );
   }
     `
+  },{
```
Comment (itaditya) [inline hooks.js]: add a space between `,` and `{`

File: hooks.js [hunk 2]
```diff
@@ -151,5 +151,51 @@ export const hooks = [
...
+      window.matchMedia(query).matches
+    );
+  
+    useEffect(() => {
+        let media = window.matchMedia(query);
+        if (media.matches !== matches) {
+          setMatches(media.matches);
+        }
+        let listener = () =>
+          setMatches(media.matches);
```
Comment (itaditya) [inline hooks.js]: Put these two lines into a single line

```js
let listener = () => setMatches(media.matches);
```

**Comment by itaditya** [comment]
Just two minor comments. Rest everything looks great!

**Comment by itaditya** [comment]
@JHughes11 everything looks great. Squash the commits and force push it here. I'll merge it ASAP

**Comment by JHughes11** [comment]
Should be good now! First time I've squashed before so hopefully I did it right.

**Comment by itaditya** [comment]
It's great that you got to try something new with this PR. Thanks for working on this @JHughes11 

Congrats

**Pull request merged**